src/plugins/discover/public/application/components/results_action_bar/results_action_bar.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import './results_action_bar.scss';
+
+import React from 'react';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { HitsCounter } from '../chart/hits_counter';
+import { OpenSearchSearchHit } from '../../doc_views/doc_views_types';
+import { DiscoverOptions } from '../discover_options/discover_options';
+
+export interface DiscoverResultsActionBarProps {
+  hits?: number;
+  showResetButton?: boolean;
+  resetQuery(): void;
```

Comment:
nit: should we make this optional since it is only used when showResetButton is true

Reply:
While i agree with your statement, I wouldn't suggest we go that route for the following reasons:

- It was previously implemented as a required prop (see `HitsCounter` component)
- `ResultsActionBar` only is used from `canvas/index.tsx`, and from there we always pass the prop
- Having both `showResetButton` and `resetQuery` as optionals now adds unwanted complexities. For example, what if `showResetButton` is true while `resetQuery` is not defined? In an ideal situtation, `resetQuery()` should be optional and `showResetButton` shouldn't exist. The component should simply render a reset button if `resetQuery` is defined. But I feel that is unnecessary work and current implementation works fine

Reply:
actually i change my mind, lets do my last bullet point :D